src/www/app/outline_server_repository/access_key_serialization.ts

```diff
@@ -31,6 +31,52 @@ export function staticKeyToShadowsocksSessionConfig(staticKey: string): Shadowso
       password: config.password.data,
     };
   } catch (error) {
-    throw new errors.ServerUrlInvalid(error.message || 'Failed to parse static access key.');
+    throw new errors.ServerAccessKeyInvalid(error.message || 'Failed to parse static access key.');
+  }
+}
+
+// fetches information from a dynamic access key and attempts to parse it
+// TODO(daniellacosse): unit tests
+export async function promiseShadowsocksSessionConfigFromDynamicAccessKey(
```

Comment:
async functions already imply a Promise value and we don't use a " promise" prefix for async functions.

Reply:
Perhaps just call `fetchShadowsocksSessionConfig(configLocation)`?